guides/keras_nlp/transformer_pretraining.py

```diff
@@ -0,0 +1,481 @@
+"""
+
+# Load SST-2.
+sst_train_ds = tf.data.experimental.CsvDataset(
+    "SST-2/train.tsv", [tf.string, tf.int32], header=True, field_delim="\t"
+).batch(FINETUNING_BATCH_SIZE)
+sst_val_ds = tf.data.experimental.CsvDataset(
+    "SST-2/dev.tsv", [tf.string, tf.int32], header=True, field_delim="\t"
+).batch(FINETUNING_BATCH_SIZE)
+
+# Load wikitext-103 and filter out short lines.
+wiki_train_ds = (
+    tf.data.TextLineDataset("wikitext-103-raw/wiki.train.raw")
+    .filter(lambda x: tf.strings.length(x) > 100)
+    .batch(PRETRAINING_BATCH_SIZE)
+)
+wiki_val_ds = (
+    tf.data.TextLineDataset("wikitext-103-raw/wiki.valid.raw")
+    .filter(lambda x: tf.strings.length(x) > 100)
+    .batch(PRETRAINING_BATCH_SIZE)
+)
+
+# Take a peak at the sst-2 dataset.
+list(sst_train_ds.unbatch().take(4).as_numpy_iterator())
+
+"""
+You can see that our `SST-2` dataset contains relatively short snippets of movie review
+text. Our goal is to predict the sentiment of the snippet. A label of 1 indicates
+positive sentiment, and a label of 0 negative sentiment.
+"""
+
+"""
+### Establish a baseline
+
+As a first step, we will establish a baseline of good performance. We don't actually need
+KerasNLP for this, we can just use core Keras layers.
+
+We will build a simple bag-of-words model, where we learn a positive or negative weight
+for each word in our input. A sample's score is simply the sum of all words that are
+present in the sample.
+"""
+
+# This layer will turn our input sentence into a list of 1s and 0s the same size
+# our vocabulary, indicating whether a word is present in absent.
+multi_hot_layer = keras.layers.TextVectorization(
+    max_tokens=4000, output_mode="multi_hot"
+)
+multi_hot_layer.adapt(sst_train_ds.map(lambda x, y: x))
+# We then  learn a linear regression over that layer, and that's our entire
+# baseline model!
+regression_layer = keras.layers.Dense(1, activation="sigmoid")
+
+inputs = keras.Input(shape=(), dtype="string")
+outputs = regression_layer(multi_hot_layer(inputs))
+baseline_model = keras.Model(inputs, outputs)
+baseline_model.compile(loss="binary_crossentropy", metrics=["accuracy"])
+baseline_model.fit(sst_train_ds, validation_data=sst_val_ds, epochs=5)
+
+"""
+A bag-of-words approach can be a fast and suprisingly powerful, especially when input
+examples contain a large number of words. With shorter sequences, it can hit a
+performance ceiling.
+
+To do better, we would like to build a model that can evaluate words *in context*. It's
+more useful to see the phrase "not my cup of tea" in an input, than to be told a sample
+contained the word "cup" or "tea" individually. Instead of evaluating each word in a
+void, we need to use the information contained in the *entire ordered sequence* of our
+input.
+
+This runs us into a problem. `SST-2` is very small dataset, and there's simply not enough
+example text to attempt to build a larger, more parameterized, models that can learn on a
```

Comment:
nit: remove ',' before "models that...". 

Also "models" should be in the singular form.